src/simulators/matrix_product_state/matrix_product_state.hpp

```diff
@@ -594,6 +598,22 @@ void State::snapshot_state(const Operations::Op &op,
   result.data.add_pershot_snapshot("statevector", op.string_params[0], statevector);
 }
 
+void State::snapshot_amplitudes(const Operations::Op &op,
+				ExperimentResult &result,
+				std::string name) {
+  if (op.params_amplitudes.empty()) {
+    throw std::invalid_argument("Invalid amplitudes snapshot (No base value given).");
+  }
+  reg_t base_values;
+  for (const auto &param : op.params_amplitudes) {
+    base_values.push_back(param);
+  }
+  uint_t num_amplitudes = base_values.size();
+  cvector_t amplitude_vector(num_amplitudes);
+  qreg_.get_amplitude_vector(base_values, amplitude_vector);
```

Comment:
A small comment here. I think it is better to move the creation of `amplitude_vector` inside the function `get_amplitude_vector` and return it. Like this:
```
auto amplitude_vector = qreg_.get_amplitude_vector(base_values);
```
This way the function `get_amplitude_vector` is fully in charge of creating that vector and someone using the function doesn't need to care about creating the vector, etc.

Reply:
I agree. Done.